Frequency/Board/Map/Map.py

```diff
@@ -33,24 +35,31 @@ def GenerateTiles(self):
     def DetermineTileType(self, X, Y):
         if X < 7 and Y < 7:
             return ForestTile
-        if X > 12 and X < 20 and Y < 7:
+        if X > 10 and X < 18 and Y < 7:
             return IceTile
-        if X < 7 and Y >12 and Y < 20:
+        if X < 7 and Y >10 and Y < 18:
             return DesertTile
-        if X > 12 and X < 20 and Y > 12:
+        if X > 10 and X < 18 and Y > 10:
             return SwampTile
-        if X > 7 and X < 12 and Y > 7 and Y < 12:
+        if X > 6 and X < 11 and Y > 6 and Y < 11:
             return GoldTile
         else:
             return SeaTile
 
 
-
-
     def Update(self, game):
-        return Map(self.Resolution, [tile.Update(game) for tile in self.Tiles])
+        nList = []
+        for row in self.Tiles:
+            nRow = []
+            for tile in row:
+                tile.Update(game)
+                TileType = self.DetermineTileType(tile.Position.X, tile.Position.Y)
+                nRow.append(TileType(Vector2(row, tile), Vector2(tile.Size.X, tile.Size.Y)))
+            nList.append(nRow)
+        return nList
```

Comment:
wierd construction! Probably it gives the right solution. But why creating an new Tile. if tile.update already creating the right new Tile